modules/jwplayerTargeting.js

```diff
@@ -0,0 +1,147 @@
+import { config } from '../src/config.js';
+import { ajaxBuilder } from '../src/ajax.js';
+import { logError, isPlainObject } from '../src/utils.js';
+import { getGlobal } from '../src/prebidGlobal.js';
+import { module } from '../src/hook.js';
+
+const segCache = {};
+let requestCount = 0;
+let requestTimeout;
+let resumeBidRequest;
+
+config.getConfig('jwTargeting', (config) => {
+  // fetch media ids
+  fetchTargetingInformation(config.jwTargeting)
+});
+
+getGlobal().requestBids.before(ensureFeedRequestCompletion);
+
+export function fetchTargetingInformation(jwTargeting) {
+  const mediaIDs = jwTargeting.mediaIDs;
+  requestCount = mediaIDs.length;
+  mediaIDs.forEach(mediaID => {
+    fetchTargetingForMediaId(mediaID);
+  });
+}
+
+export function ensureFeedRequestCompletion(requestBids, bidRequestConfig) {
+  if (requestCount <= 0) {
+    requestBids.apply(this, [bidRequestConfig]);
+    return;
+  }
+  resumeBidRequest = requestBids.bind(this, bidRequestConfig);
```

Comment:
What is `this` in this context? `window`?

Reply:
it's the same context as the original `getGlobal().requestBids` that we blocked.